hazelcast/src/main/java/com/hazelcast/multimap/impl/MultiMapContainerSupport.java

```diff
@@ -66,7 +71,146 @@ public MultiMapValue getMultiMapValueOrNull(Data dataKey) {
         return multiMapValues;
     }
 
+    public void setValue(Data dataKey, MultiMapValue value) {
+        MultiMapValue prevValue = multiMapValues.put(dataKey, value);
+        if (prevValue != null) {
+            decrSize(prevValue.size());
+        }
+        incSize(value.size());
+    }
+
+    public boolean setValue(Data dataKey, Collection<MultiMapRecord> records) {
+        MultiMapValue multiMapValue = getOrCreateMultiMapValue(dataKey);
+        Collection<MultiMapRecord> mapRecords = multiMapValue.getCollection(false);
+        decrSize(mapRecords.size());
+        mapRecords.clear();
+
+        boolean added = mapRecords.addAll(records);
+        if (added) {
+            incSize(records.size());
+        }
+        return added;
+    }
+
+    public boolean addValue(Data dataKey, MultiMapRecord value) {
+        return addValue(dataKey, value, -1);
+    }
+
+    public boolean addValue(Data dataKey, MultiMapRecord value, int index) {
+        MultiMapValue multiMapValue = getOrCreateMultiMapValue(dataKey);
+        boolean added;
```

Comment:
Its nitpicking and mostly personal taste, but I find it difficult to follow the boolean changes in the code, and more error prone. How about removing it, and having multiple method-exit points (`return`). eg. `return multiMapValue.getCollection(false).add(value);`